prov/efa/src/efa_domain.c

```diff
@@ -494,53 +562,15 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 
 	domain->cache = NULL;
 
-	/*
-	 * Call ibv_fork_init if the user asked for fork support.
-	 */
-	if (efa_fork_status == EFA_FORK_SUPPORT_ON) {
-		ret = -ibv_fork_init();
-		if (ret) {
-			EFA_WARN(FI_LOG_DOMAIN,
-			         "Fork support requested but ibv_fork_init failed: %s\n",
-			         strerror(-ret));
-			goto err_free_info;
-		}
-	}
-
-	/*
-	 * Run check to see if fork support was enabled by another library. If
-	 * one of the environment variables was set to enable fork support,
-	 * this variable was set to ON during provider init.  Huge pages for
-	 * bounce buffers will not be used if fork support is on.
-	 */
-	if (efa_fork_status == EFA_FORK_SUPPORT_OFF &&
-	    efa_check_fork_enabled(*domain_fid))
-		efa_fork_status = EFA_FORK_SUPPORT_ON;
+#ifndef _WIN32
+	ret = efa_initialize_fork_support(*domain_fid);
```

Comment:
I am not familiar with windows. Does windows use the Copy on Write mechanism to implement fork?

Reply:
Currently there is no path to using fork on Windows reliably in all situations - especially those applications that rely on driver interaction like the efa provider.  There are implementations of fork for Windows that vary in effectiveness, but the overhead and maintenance of such implementations are fairly large.  This will be documented as not supported on Windows for efa provider.